lib/schema.js

```diff
@@ -668,7 +671,20 @@ class Schema {
 	 * @return {Object}
 	 */
 	toJSONSchema() {
-		return this._subschemaToJSONSchema(this.getData());
+		// Used to register JSON Schema definitions at the root of the schema.
+		this.jsonSchemaDefinitions = {};
+
+		let data = this.getData();
+		let jsonSchema = this._subschemaToJSONSchema(data);
+
+		// Attach definitions, if any
+		if (Object.keys(this.jsonSchemaDefinitions).length) {
```

Comment:
It looks like `this.jsonSchemaDefinitions` is always empty by this point, as it is set above to `{}`.  `#getData` and `#_subschemaToJSONSchema` do not have side effects from what I can tell, so it would never get populated.